libsolidity/analysis/ControlFlowRevertPruner.cpp

```diff
@@ -83,14 +83,19 @@ void ControlFlowRevertPruner::findRevertStates()
 					if (resolvedFunction == nullptr || !resolvedFunction->isImplemented())
 						continue;
 
-					switch (m_functions.at({findScopeContract(*resolvedFunction, item.contract), resolvedFunction}))
+					CFG::FunctionContractTuple calledFunctionTuple{
+						findScopeContract(*resolvedFunction, item.contract),
+						resolvedFunction
+					};
+					switch (m_functions.at(calledFunctionTuple))
 					{
 						case RevertState::Unknown:
+							wakeUp[calledFunctionTuple].insert(item);
 							foundUnknown = true;
 							return;
 						case RevertState::AllPathsRevert:
 							return;
-						default:
+						case RevertState::HasNonRevertingPath:
 							break;
```

Comment:
I'd add `solAssert(false, "")` in the default case just in case.

Reply:
You mean I should add a default case with a failing assertion?

Reply:
Yes. In case something is added to the enum and not handled here. I guess the risk of that here is pretty low but personally I'd add it anyway.

Reply:
The C++ compiler will create an error in that case.

Reply:
If we add a default case, it will not create an error and we only notice it at runtime.

Reply:
Oh, looks like it does. That's even better. It's fine then.